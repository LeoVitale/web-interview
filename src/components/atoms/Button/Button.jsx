import React from 'react'
import PropTypes from 'prop-types'
import { button, link } from './Button.module.scss'

const Button = ({ children, className, type, ...props }) => {
  const style = type === 'link' ? link : button
  return (
    <button className={`${style} ${className}`} {...props}>
      {children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
}

export default Button
