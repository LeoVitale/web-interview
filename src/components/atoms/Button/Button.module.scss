@import 'styles/_constants.scss';

.button {
  cursor: pointer;
  display: block;
  width: 100%;
  height: 40px;
  color: white;
  background-color: $positive-light;
  font-size: 1em;
  border-radius: 8px;
  border: none;
  &:hover {
    background-color: $positive;
  }
}

.disabledButton {
  cursor: initial;
  background-color: $hover-grey;
  color: $icon-color;
  &:hover {
    background-color: $hover-grey;
  }
}

.link {
  cursor: pointer;
  color: $positive;
  background: transparent;
  border: none;
}
