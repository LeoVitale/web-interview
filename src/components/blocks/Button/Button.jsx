import React from 'react';
import styles from './Button.module.scss';

const Button = ({ children, ...props}) => {
  const { button } = styles;
  return (
    <button className={button} {...props}>
      {children}
    </button>
  );
};

export default Button;
