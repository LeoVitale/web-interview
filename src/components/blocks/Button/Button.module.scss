@import 'styles/_constants.scss';

.button {
  cursor: pointer;
  display: block;
  width: 100%;
  height: 40px;
  color: white;
  background-color: $positive-light;
  font-size: 1.4em;
  border-radius: 8px;
  &:hover {
    background-color: $positive;
  }
}
