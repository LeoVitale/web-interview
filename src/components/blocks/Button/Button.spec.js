import React from 'react';
import { mount } from 'enzyme';
import Button from './Button.jsx';
import Index from '.';

describe('<Button />', () => {
  const button = mount(<Button />);

  it('test index.js import', () => {
    expect(Index).toHaveLength(0);
  });

  it('should render Button as expected', () => {
    expect(button).toMatchSnapshot();
  });
});
