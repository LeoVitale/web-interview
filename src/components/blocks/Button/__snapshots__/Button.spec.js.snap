// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render Button as expected 1`] = `
<Button>
  <div>
    Button
  </div>
</Button>
`;
